File: day-1/c/day-1.c
#include <stdio.h>
#include <stdlib.h>
#include <libgen.h>


int main( int argc, char *argv[] ) {

    FILE *f;
    int cur_depth, next_depth;
    int cnt_increases = 0;

    // Assumes first cmdline param will be input filename. If not passed, print usage and exit
    if (argc < 2) {
        printf("ERROR: Missing input file\n");
        printf("       Usage %s input-file-name\n", basename(argv[0]));

        exit(EXIT_FAILURE);
    }

    // Open the file for reading
    if ((f = fopen(argv[1], "r")) == NULL) {
        printf( "ERROR: failed to open %s\n", argv[1]) ;
        exit(EXIT_FAILURE);
    }
    
    // Read the first depth;
    fscanf(f, "%d", &cur_depth);

    // Loop thru rest
    while (fscanf(f, "%d", &next_depth) != EOF) {
        if (cur_depth < next_depth) // Check if we increased
            cnt_increases++;
        cur_depth = next_depth ;
    }

    // cleanup 
    fclose(f);

    // Print the result
    printf( "Number of increases %d\n", cnt_increases) ;

}

File: day-1/python/day-1.py
#!/usr/bin/env python3

import sys


def main(argv): 

    # input file is supposed to be the first param - check that something was passed in
    if ( len( argv ) > 1 ):
        input_file = argv[1]
    else:
        print( "ERROR: Missing input file" )
        print( "       Usage {} input-file-name\n".format(argv[0].split('/')[-1]) )
        exit(1)

    # Initializations
    cur_depth = sys.maxsize 
    cnt_increases = 0

    # Open file and process line by line (better for large files)
    with open( input_file ) as f:
        for line in f: # Read a line
            next_depth = int( line.strip() ) # Convert string to integer

            if ( next_depth > cur_depth ): # Check if increase
                cnt_increases += 1

            cur_depth = next_depth
        
    f.close( )

    print( "Number of Increases {}".format(cnt_increases) )





if __name__ == "__main__":
   main(sys.argv)